mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergFilterFactory.java

```diff
@@ -176,4 +179,20 @@ private static int daysFromTimestamp(Timestamp timestamp) {
   private static long microsFromTimestamp(Timestamp timestamp) {
     return DateTimeUtil.microsFromInstant(timestamp.toInstant());
   }
+
+  private static Class getSearchArgumentImplClass() {
+    Class<?>  predicateLeafImplClass;
+    try {
+      // SearchArgumentImpl is a package-private class in Hive 1.1, but not in Hive > 2
+      predicateLeafImplClass = Stream.of(
+          Class.forName("org.apache.hadoop.hive.ql.io.sarg.SearchArgumentImpl").getDeclaredClasses())
+          .filter(subClass -> "PredicateLeafImpl".equals(subClass.getSimpleName()))
```

Comment:
Can we just use DynClasses here to find `org.apache.hadoop.hive.ql.io.sarg.SearchArgumentImpl$PredicateLeafImpl`?

Reply:
Thanks for the advice, I didn't know we had DynClasses and it could be used this way. Tested on Hive1.1 too